### ISSUE TITLE
please support ragged tensor as memory for AttentionMechanism

### ISSUE THREAD

**Issue by breadbread1984** [tensorflow/addons#2371]
**Describe the feature and the current behavior/state.**

AttentionMechanism doesn't support ragged tensor for memory. the output hidden sequences of tf.keras.layers.RNN already support ragged tensor. if AttensionMechanism support ragged tensor it would be very convenient.

**Relevant information**
- Are you willing to contribute it (yes/no): no
- Are you willing to maintain it going forward? (yes/no): no
- Is there a relevant academic paper? (if so, where):
- Is there already an implementation in another framework? (if so, where):
- Was it part of tf.contrib? (if so, where):

**Which API type would this fall under (layer, metric, optimizer, etc.)**

the improvement doesn't require any new APIs

**Who will benefit with this feature?**

NLP develop using ragged tensor as input

**Any other info.**

**Comment by guillaumekln** [comment]
Thanks for the feature request. It makes sense to support ragged tensors but they should ideally be supported in all modules that accept variable length inputs, and not just the attention memory.

Are you willing to contribute?

**Comment by breadbread1984** [comment]
I would like to, but I am not competent to complish this task.

**Comment by seanpmorgan** [comment]
TensorFlow Addons is transitioning to a minimal maintenance and release mode. New features will not be added to this repository. For more information, please see our public messaging on this decision:
[TensorFlow Addons Wind Down](https://github.com/tensorflow/addons/issues/2807)

Please consider sending feature requests / contributions to other repositories in the TF community with a similar charters to TFA:
[Keras](https://github.com/keras-team/keras)
[Keras-CV](https://github.com/keras-team/keras-cv)
[Keras-NLP](https://github.com/keras-team/keras-nlp)
